GVFS/GVFS.UnitTests/Service/MacServiceTests.cs

```diff
@@ -0,0 +1,141 @@
+﻿using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using GVFS.Service;
+using GVFS.UnitTests.Mock.Common;
+using GVFS.UnitTests.Mock.FileSystem;
+using Moq;
+using NUnit.Framework;
+using System.IO;
+
+namespace GVFS.UnitTests.Service
+{
+    [TestFixture]
+    [NonParallelizable]
+    public class MacServiceTests
+    {
+        private const string GVFSServiceName = "GVFS.Service";
+        private const string ActiveUserId = "502";
+        private static readonly string RegisteredRepo = Path.Combine("mock:", "code", "repo2");
+        private static readonly string ServiceDataLocation = Path.Combine("mock:", "registryDataFolder");
+
+        private MockFileSystem fileSystem;
+        private MockTracer tracer;
+        private Mock<MockPlatform> gvfsPlatformMock;
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.tracer = new MockTracer();
+            this.fileSystem = new MockFileSystem(new MockDirectory(ServiceDataLocation, null, null));
+            this.gvfsPlatformMock = new Mock<MockPlatform>();
+            this.gvfsPlatformMock.Setup(p => p.GetCurrentUser()).Returns(ActiveUserId);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            this.fileSystem = null;
+            this.tracer = null;
+        }
+
+        [TestCase]
+        public void ServiceStartTriggersAutoMountForCurrentUser()
+        {
+            Mock<RepoRegistry> repoRegistry = new Mock<RepoRegistry>();
+            repoRegistry.Setup(r => r.AutoMountRepos(It.IsAny<int>()));
+
+            GVFSService service = new GVFSService(
+                this.tracer,
+                GVFSServiceName,
+                startListening: false,
+                repoRegistry: repoRegistry.Object,
+                gvfsPlatform: this.gvfsPlatformMock.Object);
+
+            service.RunWithArgs(new string[] { $"--servicename={GVFSServiceName}" });
+
+            int expectedUserId = int.Parse(ActiveUserId);
+
+            repoRegistry.Verify(
+                r => r.AutoMountRepos(It.Is<int>(arg => arg == expectedUserId)),
+                Times.Once,
+                $"{nameof(repoRegistry.Object.AutoMountRepos)} was not called during Service startup");
+
+            repoRegistry.Verify(
+                r => r.TraceStatus(),

```

Comment:
I like the first assert, since that basically says "automount is called during startup". This one I'm not sure what value we're getting. It feels like we're asseting on implementation details?

Reply:
Agree. Removed `TraceStatus` tracking.